### DISCUSSION TITLE
Semantic API state mismatch in @Order

### DISCUSSION THREAD

**Discussion by skohlmann** [junit-team/junit-framework#2909] [General]
Hi,

the [`@Order`](https://github.com/junit-team/junit5/blob/r5.8.2/junit-jupiter-api/src/main/java/org/junit/jupiter/api/Order.java) annotation is marked as `EXPERIMENTAL` where the type annotation value `OrderAnnotation` implementation is not marked as `EXPERIMENTAL`. So I guess, it‘s stable. The documentation of `OrderAnnotation` also refers to `@Order.` 

On the other hand, if `OrderAnnotation` implicitly takes the API state of `@Order`, this indirection is semantically difficult to detect. This also in view of the fact that `@Order` has been in the `EXPERIMENTAL` state for many versions, and within this time period certainly

1. a realization with regard to 👍 or 👎 has arisen.
2. the use in projects is certainly established, after 4 minor releases.

I would be happy if the situation is clarified by the team in the next minor release. Whereby I plead for `@Order` to be transferred to the state `STABLE`.

Have fun :-) 

-sascha

**Comment by sbrannen** [comment]
`org.junit.jupiter.api.MethodOrderer.OrderAnnotation` is no longer experimental; however, `org.junit.jupiter.api.ClassOrderer.OrderAnnotation` is still experimental.

In any case, I think your argument is valid: the `@Order` annotation itself should no longer be experimental.

**Comment by sbrannen** [comment]
I've created #2910 to address this.

**Comment by sbrannen** [comment]
`@Order` will be a stable API as of JUnit Jupiter 5.9 M1.

**Reply by skohlmann** [reply]
Many thanks for the fast solution. Great job.
